src/digests.rs

```diff
@@ -0,0 +1,211 @@
+//! # Multihash
+//!
+//! Implementation of [multihash](https://github.com/multiformats/multihash) in Rust.
+//!
+//! A `Multihash` is a structure that contains a hashing algorithm, plus some hashed data.
+//! A `MultihashRef` is the same as a `Multihash`, except that it doesn't own its data.
+//!
+
+use std::convert::TryFrom;
+
+use bytes::{BufMut, Bytes, BytesMut};
+use num_traits::cast::FromPrimitive;
+use unsigned_varint::{decode as varint_decode, encode as varint_encode};
+
+use crate::errors::{DecodeError, DecodeOwnedError};
+use crate::hashes::Code;
+
+/// Represents a valid multihash.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Multihash {
+    bytes: Bytes,
+}
+
+impl Multihash {
+    /// Verifies whether `bytes` contains a valid multihash, and if so returns a `Multihash`.
+    pub fn from_bytes(bytes: Vec<u8>) -> Result<Multihash, DecodeOwnedError> {
+        if let Err(err) = MultihashRef::from_slice(&bytes) {
+            return Err(DecodeOwnedError {
+                error: err,
+                data: bytes,
+            });
+        }
+        Ok(Multihash {
```

Comment:
As far as I understand it `MultihashRef` is dropped here but we know `bytes` are valid or we'd have `Err`. I do however find it confusing / awkward, probably my lack of  Rust experience, makes me react with [Parse don't validate](https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/)

I think parse approach would not work here as `MultihashRef` contains  `&[u8]`, but maybe it make sense to have separate fn that does not construct `MultihashRef` ?

Maybe this is idiomatic in Rust, but maybe a comment for newbies (like myself) would be worth it.